### ISSUE TITLE
Add unit test for jittability/autodiffability of `Agent` construction

### ISSUE THREAD

**Issue by conorheins** [infer-actively/pymdp#295]
In model inversion workflows with [pybefit](https://github.com/dimarkov/pybefit) and [numpyro](https://github.com/pyro-ppl/numpyro), a user will often have to write a custom parameterization function `transform()` function that takes values of a latent variable (e.g., samples) and uses them to parameterize a pymdp agent with those samples, e.g.,

```python
def transform(z):
    A, B, C, D, E = parameterize_model_from_samples(z) # some code here that parameterizes the probability tensors using samples
    return Agent(A, B, C, D, E)
```

Since this function will be called iteratively in the course of a NUTS / MCMC loop or a stochastic variational inference loop, the `Agent` constructor (`__init__`) needs to be autodifferentiable and jittable. We should have a unit test that ensures this, e.g.

```python

def test_agent_construction_jittability(...)
     def construct_agent(params):
            return Agent(**params)
     jit(construct_agent)(params) # make sure this runs
```

and likewise one that returns well-defined/non-NaNed gradients e.g. ensuring `~jnp.any(jnp.isnan(jax.grad(construct_agent)(params)))`

**Comment by conorheins** [comment]
The ideal unit test here should also test the differentiability of the one-step active inference process for a set of plausible different configurations of the `Agent` class (different settings of learning and inference algorithms), to make sure all those sub-processes are also autodifferentiable.
